users/models.py

```diff
@@ -1,3 +1,290 @@
 from django.db import models
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager, PermissionsMixin, Group, Permission
+from django.core.validators import EmailValidator, RegexValidator
+from django.core.exceptions import ValidationError
+from django.utils import timezone
+from django.contrib.auth.password_validation import validate_password
+import logging
+import json
 
-# Create your models here.
+logger = logging.getLogger(__name__)
+
+
+class CustomUserManager(BaseUserManager):
+    """Custom user manager for the User model."""
+
+    def create_user(self, email, password=None, **extra_fields):
+        """
+        Create and save a new user with the given email and password.
+
+        Args:
+            email (str): User's email address.
+            password (str, optional): User's password.
+            **extra_fields: Additional fields for the user.
+
+        Returns:
+            User: Newly created user instance.
+
+        Raises:
+            ValueError: If email is not provided or password is invalid.
+        """
+        if not email:
+            raise ValueError('The Email field must be set')
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        if password:
+            if not self.validate_password(password):
+                raise ValueError(
+                    'Password must be at least 8 characters long and contain at least one uppercase letter, one lowercase letter, one digit, and one special character.')
+            try:
+                validate_password(password)
+                user.set_password(password)
+            except ValidationError as e:
+                raise ValueError(f"Invalid password: {', '.join(e.messages)}")
+        else:
+            raise ValueError('Password must be provided')
```

Comment:
Remove the manual validation and rely fully on Django’s `validate_password()` function, which supports adding custom validators through the Django settings.